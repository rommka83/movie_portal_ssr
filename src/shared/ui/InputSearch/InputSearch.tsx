import { useState, ChangeEvent, useEffect } from 'react';
import styles from './inputsearch.module.css';
import classNames from 'classnames';

interface IInputSearch {
  placeholderText: string;
  className?: string;
  onChange: (value: string) => void;
  searchValue: string;
  onSearchClick: () => void;
}
export function InputSearch({
  placeholderText,
  className,
  onChange,
  searchValue,
  onSearchClick,
}: IInputSearch) {
  const [value, setValue] = useState('');

  const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
    const value = event.currentTarget.value;
    setValue(value);
    onChange(value);
  };
  useEffect(() => {
    if (searchValue) {
      setValue(searchValue);
    }
  }, [searchValue]);
  return (
    <div>
      <label className={styles.inputContainer}>
        <input
          className={classNames({ className }, styles.inputSearch)}
          type="text"
          onChange={onChangeHandler}
          value={value}
          placeholder=" "
        />
        <button
          className={classNames('icon-search_20__0', styles.searchButton, {
            [styles.activeIcon]: searchValue,
          })}
          disabled={!searchValue}
          onClick={onSearchClick}
        />
        <span className={styles.textPlaceholder}>{placeholderText}</span>
      </label>
    </div>
  );
}
