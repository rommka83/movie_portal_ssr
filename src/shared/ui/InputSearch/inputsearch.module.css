.inputContainer {
  position: relative;
}

.inputSearch {
  height: 40px;
  width: 270px;
  border-radius: 8px;
  color: var(--dark-gray);
  background-color: var(--gray-7e0);
  padding: 18px 40px 2px 12px;
  cursor: pointer;
  font-size: 15px;
  line-height: 20px;
  font-weight: 500;
}

.inputSearch:focus ~ .icon {
  color: var(--gray-light-32);
}

.icon {
  position: absolute;
  top: -5px;
  right: 12px;
  font-size: 16px;
  line-height: 1;
  color: var(--gray-32);
}

.textPlaceholder {
  position: absolute;
  top: -7px;
  left: 9px;
  color: var(--gray-light);
  font-size: 15px;
  line-height: 20px;
  pointer-events: none;
  transition: all 150ms linear;
}

.inputSearch:focus ~ .textPlaceholder {
  font-size: 12px;
  line-height: 1;
  top: -15px;
}
