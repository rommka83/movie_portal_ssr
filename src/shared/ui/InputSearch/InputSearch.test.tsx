import renderer from 'react-test-renderer';
import { InputSearch } from './InputSearch';

describe('проверка InputSearch', () => {
  test('отрисовка InputSearch', () => {
    const component = renderer.create(
      <InputSearch
        placeholderText='Проверка'
        onChange={function (value: string): void {
          throw new Error('Function not implemented.');
        }}
        searchValue={''}
        onSearchClick={function (): void {
          throw new Error('Function not implemented.');
        }}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
