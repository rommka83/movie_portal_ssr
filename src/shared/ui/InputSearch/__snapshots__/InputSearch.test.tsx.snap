// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка InputSearch отрисовка InputSearch 1`] = `
<div>
  <label
    className="inputContainer"
  >
    <input
      className="inputSearch"
      onChange={[Function]}
      placeholder=" "
      type="text"
      value=""
    />
    <button
      className="icon-search_20__0 searchButton"
      disabled={true}
      onClick={[Function]}
    />
    <span
      className="textPlaceholder"
    >
      Проверка
    </span>
  </label>
</div>
`;
